### DISCUSSION TITLE
Weird behaviour with MultiSelect

### DISCUSSION THREAD

**Discussion by mattfullerdev** [mantinedev/mantine#1347] [Q&A] [answered]
I'm trying to use MultiSelect to bring in data from an API.

Here's the component:

```
export const AddThings = () => {
    const [data, setData] = useState<{value: string, label: string}[]>([]);

    useEffect(() => {
        apiService.getThings('').then((res) => setData(res.data));
    }, []);

    const searchThings= (searchTerm: string) => {
        apiService.getThings(searchTerm).then((res) => setData(res.data));
    };

    return (
        <>
            <Text>Add Things</Text>
            <MultiSelect
                data={data}
                label="Choose things"
                placeholder="Pick all that you like"
                searchable
                onSearchChange={(e) => searchThings(e)}
                limit={10}
            />
        </>
    );
}
```
Initial load brings in the first 10 items, choosing from this list works fine and the values added as expected. Then if you search, it clears the already selected values, you can still add values from these new results but with the previous items missing from the UI. The state still holds the complete list of selected values however. The limit is set on both front and back end.

Have I missed something?

**Comment by rtivital** [comment, marked as answer]
MultiSelect values are infered from data, it won't be able to display them if they are not included.

```tsx
<MultiSelect data={[{ value: 'first', label: 'Hello' }]} value={['second', 'first']} />
```
This example will be able to display item with value `first` but now with value `second` because there is no way to get item label. Make sure that you include data for all selected items in your `data` array.

**Reply by mattfullerdev** [reply]
Thanks, that did it. Pulled in all the data from the API and it works as it should.
